src/QMCHamiltonians/BareKineticEnergy.h

```diff
@@ -18,9 +18,15 @@
 #define QMCPLUSPLUS_BAREKINETICENERGY_H
 
 #include "QMCHamiltonians/OperatorBase.h"
+#include <ResourceCollection.h>
+#include <ResourceHandle.h>
 
 namespace qmcplusplus
 {
+
+template<typename T>
+class ListenerVector;
```

Comment:
The full header file has been included in the base class. So why this is needed?

If the intention is to reduce header file inclusion. Then base class file can carry forward declaration instead of here.